ephios/core/signup/forms.py

```diff
@@ -32,27 +38,52 @@ def clean_individual_end_time(self):
             return None
         return self.cleaned_data["individual_end_time"]
 
+    def get_comment_visibility(self):
+        return (
+            ParticipationComment.Visibility.PUBLIC
+            if self.cleaned_data["comment_is_public"]
+            else ParticipationComment.Visibility.PARTICIPANT
+        )
+
     def clean(self):
         cleaned_data = super().clean()
+        cleaned_data["comment_visibility"] = self.get_comment_visibility()
         if not self.errors:
             start = cleaned_data["individual_start_time"] or self.shift.start_time
             end = cleaned_data["individual_end_time"] or self.shift.end_time
             if end < start:
                 self.add_error("individual_end_time", _("End time must not be before start time."))
             return cleaned_data
 
+    def save(self):
```

Comment:
hmm we can't remove the commit parameter here, as e.g. the named teams Participation Form calls save with the commit parameter. I actually ran into the exception while testing.